### ISSUE TITLE
Add support for non-blocking ("async") YAML parsing

### ISSUE THREAD

**Issue by guillermocalvo** [FasterXML/jackson-dataformats-text#165]
Jackson-core supports [non-blocking parsers](http://fasterxml.github.io/jackson-core/javadoc/2.10/com/fasterxml/jackson/core/JsonFactory.html#createNonBlockingByteArrayParser--) since [version 2.9](http://fasterxml.github.io/jackson-core/javadoc/2.9/com/fasterxml/jackson/core/JsonFactory.html#createNonBlockingByteArrayParser--), but YAML doesn't seem to have a non-blocking parser yet.

This is preventing us from parsing YAML request bodies directly in reactive web apps (Spring WebFlux).

These are the steps to reproduce this problem:
1. Create a YAML decoder:
```java
public class YamlDecoder extends AbstractJackson2Decoder {
    public YamlDecoder() {
        super(new YAMLMapper(), MediaType.valueOf("application/x-yaml"));
    }
}
```
2. Add it as a custom decoder:
```java
@EnableWebFlux
public class AppConfiguration implements WebFluxConfigurer {
    ...
    @Override
    public void configureHttpMessageCodecs(ServerCodecConfigurer configurer) {
        configurer.customCodecs().decoder(new YamlDecoder());
    }
}
```
3. Add a REST controller with a mapping that consumes YAML:
```java
@RestController
public class FoobarController {
    @PostMapping(value = "/foo", consumes = "application/x-yaml")
    public void foo(@RequestBody MyCustomClass body) {
        ...
    }
}
```
4. Send a YAML request. Then an `UnsupportedOperationException` will be [thrown from `JsonFactory`](https://github.com/FasterXML/jackson-core/blob/2.11/src/main/java/com/fasterxml/jackson/core/JsonFactory.java#L1122) with the message:
```
Non-blocking source not (yet?) supported for this format (YAML)
```

**Comment by cowtowncoder** [comment]
True, non-blocking handling has so far only been implemented for JSON and Smile backends.
Of other backends, CBOR should be quite easy, and XML plausible with `aalto-xml`.

But the challenge with YAML is that this would require SnakeYAML to support it, and then format module could use it.

So: if anyone has time and interest to work on this, it'd be interesting challenge. :)
